services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/bsm/OdeBsmJsonRepositoryImpl.java

```diff
@@ -0,0 +1,191 @@
+        GeodeticCalculator calculator = new GeodeticCalculator();
+        calculator.setStartingGeographicPoint(centerLng, centerLat);
+
+        calculator.setDirection(90, distance);
+        double maxLng = calculator.getDestinationGeographicPoint().getX();
+
+        calculator.setDirection(270, distance);
+        double minLng = calculator.getDestinationGeographicPoint().getX();
+
+        return new double[] { minLng, maxLng };
+    }
+
+    /**
+     * Filter OdeBsmData by originIp, vehicleId, startTime, endTime, and a bounding
+     * box
+     * 
+     * @param originIp  the origin IP
+     * @param vehicleId the vehicle ID
+     * @param startTime the start time
+     * @param endTime   the end time
+     * @param centerLng the longitude (in degrees) of the center of the bounding box
+     * @param centerLat the latitude (in degrees) of the center of the bounding box
+     * @param distance  the "radius" of the bounding box, in meters (total width is
+     *                  2x distance)
+     */
+    public List<OdeBsmData> findOdeBsmDataGeo(String originIp, String vehicleId, Long startTime, Long endTime,
+            Double centerLng, Double centerLat, Double distance) {
+        Query query = new Query();
+
+        if (originIp != null) {
+            query.addCriteria(Criteria.where("metadata.originIp").is(originIp));
+        }
+
+        if (vehicleId != null) {
+            query.addCriteria(Criteria.where("payload.data.coreData.id").is(vehicleId));
+        }
+
+        String startTimeString = Instant.ofEpochMilli(0).toString();
+        String endTimeString = Instant.now().toString();
+
+        if (startTime != null) {
+            startTimeString = Instant.ofEpochMilli(startTime).toString();
+        }
+        if (endTime != null) {
+            endTimeString = Instant.ofEpochMilli(endTime).toString();
+        }
+        query.limit(props.getMaximumResponseSize());
+        query.addCriteria(Criteria.where("metadata.odeReceivedAt").gte(startTimeString).lte(endTimeString));
+        query.fields().exclude("recordGeneratedAt");
+
+        if (centerLng != null && centerLat != null && distance != null) {
+            double[] latitudes = calculateLatitudes(centerLng, centerLat, distance);
+            double[] longitudes = calculateLongitudes(centerLng, centerLat, distance);
+
+            query.addCriteria(Criteria.where("payload.data.coreData.position.latitude")
+                    .gte(Math.min(latitudes[0], latitudes[1])).lte(Math.max(latitudes[0], latitudes[1])));
+            query.addCriteria(Criteria.where("payload.data.coreData.position.longitude")
+                    .gte(Math.min(longitudes[0], longitudes[1])).lte(Math.max(longitudes[0], longitudes[1])));
+        }
+
+        List<Map> documents = mongoTemplate.find(query, Map.class, collectionName);
+        List<OdeBsmData> convertedList = new ArrayList<>();
+
+        for (Map document : documents) {
```

Comment:
suggestion: same as https://github.com/CDOT-CV/jpo-cvmanager/pull/119#discussion_r1932693158, there seems to be several locations where we're fetching the objects as a generic form, loading the collection into memory, iterating over the collection to remove the "_id" field, manually converting to the datatype we want, and adding that to a new list before returning. This is quite a bit of extra processing and memory that we don't really need. Can we update anywhere that's using this to be more performant by either just using the class we want to fetch in the find command, or if that doesn't work excluding the "_id" field in the query object and fetching the class that way?